spring-integration-core/src/main/java/org/springframework/integration/handler/support/MessagingMethodInvokerHelper.java

```diff
@@ -943,27 +881,20 @@ private void populateHandlerMethod(Map<Class<?>, HandlerMethod> candidateMethods
 
 	@Nullable
 	private Method obtainFrameworkMethod(Class<?> targetClass) {
-		Method frameworkMethod = null;
 		for (Class<?> iface : ClassUtils.getAllInterfacesForClass(targetClass)) {
 			try {
 				// Can't use real class because of package tangle
 				if ("org.springframework.integration.gateway.RequestReplyExchanger".equals(iface.getName())) {
-					frameworkMethod =
-							ClassUtils.getMostSpecificMethod(
-									targetClass.getMethod("exchange", Message.class),
-									this.targetObject.getClass());
-					if (LOGGER.isDebugEnabled()) {
-						LOGGER.debug(this.targetObject.getClass() +
-								": Ambiguous fallback methods; using RequestReplyExchanger.exchange()");
```

Comment:
Why do we lose this debug log?

Reply:
I don’t understand why we need it at all even with the current master. Looks like we don’t do any fall backs, but just choose a first one for the `exchange` and that’s all. My feelings that it is a left over after some radical change here. So, since it doesn’t bring any value , remove it altogether!

Reply:
ok